jsettlers.logic/src/main/java/jsettlers/ai/highlevel/AiStatistics.java

```diff
@@ -277,13 +340,27 @@ private void updateFreeLand(short x, short y) {
 				sortedCuttableObjectsInDefaultPartition.put(STONE, stones);
 			}
 			stones.addNoCollission(x, y);
+			updateNearStones(x, y);
 		}
 		ELandscapeType landscape = landscapeGrid.getLandscapeTypeAt(x, y);
 		if (landscape.isRiver()) {
 			sortedRiversInDefaultPartition.addNoCollission(x, y);
 		}
 	}
 
+	private void updateNearStones(short x, short y) {
+		for (EDirection dir : EDirection.VALUES) {
+			int dx = x + dir.gridDeltaX * NEAR_STONE_DISTANCE;
```

Comment:
done.

@michaelzangl I decided to implement it with duplication of code:
```java
	public final short getNextTileX(int x) {
		return (short) (x + gridDeltaX);
	}

	public final short getNextTileY(int y) {
		return (short) (y + gridDeltaY);
	}

	public final short getNextTileX(int x, int steps) {
		return (short) (x + gridDeltaX * steps);
	}

	public final short getNextTileY(int y, int steps) {
		return (short) (y + gridDeltaY * steps);
	}
```

rather than delegation:
```java
	public final short getNextTileX(int x) {
		return getNextTileX(x, 1); 
	}

	public final short getNextTileY(int y) {
		return getNextTileY(y, 1);
	}

	public final short getNextTileX(int x, int steps) {
		return (short) (x + gridDeltaX * steps);
	}

	public final short getNextTileY(int y, int steps) {
		return (short) (y + gridDeltaY * steps);
	}
```
Because I think the extra multiplication by 1 could influence performance.
Do you agree or should I use the delegation approach?